activation/interface.go

```diff
@@ -35,7 +35,7 @@ type validatorOption func(*validatorOptions)
 
 type nipostValidator interface {
 	InitialNIPostChallengeV1(challenge *wire.NIPostChallengeV1, atxs atxProvider, goldenATXID types.ATXID) error
-	NIPostChallengeV1(challenge *wire.NIPostChallengeV1, atxs atxProvider, nodeID types.NodeID) error
+	NIPostChallengeV1(challenge *wire.NIPostChallengeV1, previous *types.ActivationTx, nodeID types.NodeID) error
```

Comment:
In my opinion it also makes sense here to just pass the wire type instead of the domain type to avoid the need for conversion.

Reply:
I initially did this, but the problem is that a golden previous ATX (after checkpoint) cannot be represented in a wire type.

Reply:
But the previous that is passed here is from a wire ATX that has been converted to a domain type ATX if I understand the code correctly? So if it is a golden ATX it would already fail before that?

Reply:
Answering myself: no, because `sql.ErrNotFound` is only returned if the blob doesn't exist. A golden ATX exists in the blobs table, it just contains no bytes.